src/dune_rules/dep_conf_eval.ml

```diff
@@ -26,16 +35,28 @@ let dep expander = function
              Build_system.Alias.dep_rec ~loc:(String_with_vars.loc s) a
            in
            [])
-  | Glob_files s ->
+  | Glob_files { glob = s; recursive } ->
     let loc = String_with_vars.loc s in
     let path = Expander.Or_exn.expand_path expander s in
     Result.map path ~f:(fun path ->
         let pred =
           Glob.of_string_exn loc (Path.basename path) |> Glob.to_pred
         in
         let dir = Path.parent_exn path in
-        Action_builder.map ~f:Path.Set.to_list
-          (File_selector.create ~dir pred |> Action_builder.paths_matching ~loc))
+        let add_dir dir acc =
+          let+ paths =
+            Action_builder.paths_matching ~loc (File_selector.create ~dir pred)
+          and+ acc = acc in
+          Path.Set.fold paths ~init:acc ~f:(fun p acc -> p :: acc)
+        in
+        let+ files =
+          let init = Action_builder.return [] in
+          if not recursive then
```

Comment:
Could this be inverted? `if not .. else ..` is awkward to read, and the length of both clauses is the same anyway.

Reply:
sure